rpc/src/rpc.rs

```diff
@@ -457,10 +500,18 @@ impl JsonRpcRequestProcessor {
         })?;
         let encoding = encoding.unwrap_or(UiAccountEncoding::Base64);
 
-        let accounts = pubkeys
-            .into_iter()
-            .map(|pubkey| get_encoded_account(&bank, &pubkey, encoding, data_slice, None))
-            .collect::<Result<Vec<_>>>()?;
+        let mut accounts = Vec::with_capacity(pubkeys.len());
+        for pubkey in pubkeys {
+            let bank = Arc::clone(&bank);
+            accounts.push(
+                self.runtime
+                    .spawn_blocking(move || {
+                        get_encoded_account(&bank, &pubkey, encoding, data_slice, None)
```

Comment:
parallelism - so that one big request doesn't monopolize the thread for too long and multiple requests can make progress together.

Reply:
I understand that point, but if we have 500 req/s, and let's assume most of them gMA don't we spend more time on spawning tasks? gMA (that can be max 100 gAI) is not the same as gPA (especially gTABO)

Reply:
>  don't we spend more time on spawning tasks

The overhead of spawning a task is minimal, it's matter of putting the task descriptor in a channel. The background thread already exists, just needs to pop the item off the channel.

The tradeoff with this approach vs doing all in one go is that with this approach, all requests potentially get a little slower (because parallelism), vs doing all in one go where some requests are faster but some risk getting starved and timeout.

Reply:
ok, then why do not use `join_all`?

Reply:
Because waiting is the point - so that if there are concurrent requests (which most likely there are), the background threads can serve those. If we join_all all the threads will fire up at the same time, load will increase, background threads will have less capacity to process other things etc.

The point of the current approach is to maximize overall throughput not single response time. I think that this is the best approach, but I'm open to changing my mind!